scripts/integration/amqp/compose.yaml

```diff
@@ -4,4 +4,23 @@ services:
   rabbitmq:
     image: docker.io/rabbitmq:${CONFIG_VERSION}
     ports:
-    - 5672:5672
+      - 5672:5672
+      - 5671:5671
+    environment:
+      - RABBITMQ_SSL_KEYFILE=/code/tests/data/ca/intermediate_server/private/rabbitmq.key.pem
+      - RABBITMQ_SSL_CERTFILE=/code/tests/data/ca/intermediate_server/certs/rabbitmq.cert.pem
+      - RABBITMQ_SSL_CACERTFILE=/code/tests/data/ca/intermediate_server/certs/ca-chain.cert.pem
+      - RABBITMQ_SSL_FAIL_IF_NO_PEER_CERT=false
+    volumes:
+      - ${PWD}:/code
+
+  wait_for_rabbitmq:
```

Comment:
Thinking about it, I'm not actually sure why this is necessary now? Does it take longer for a TLS enabled rabbitmq to start up?

If a pause is necessary, it might be easier to put the pause and run the healthcheck at the start of the test in the rust code.

Reply:
It was needed even for test w/o TLS, in old int test framework. without the wait the test will fail because rabbitmq can be not ready by the time the test runs. especially when docker build kit is enabled. It seems it is still needed in new test framework. rabbitmq image itself does not support healthcheck. I had to add the second service for that. Going to look into rs approach you mentioned.